### PR TITLE
RootLookuper: fix CNAME handling 

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

File: iterator.go [hunk 1]
```diff
@@ -233,13 +233,46 @@ func (RootLookuper) handleSuccessAnswer(_ context.Context,
...
+	exdns.ForEachRR(resp2.Answer, func(rr dns.RR) {
+		resp.Answer = append(resp.Answer, rr)
+	})
+	exdns.ForEachRR(resp2.Ns, func(rr dns.RR) {
+		resp.Ns = append(resp.Ns, rr)
+	})
+	// TODO: be smarter when merging OPTs
+	exdns.ForEachRR(resp2.Extra, func(rr dns.RR) {
+		resp.Ns = append(resp.Extra, rr)
+	})
```
Comment (karasz) [inline iterator.go:271]: This is where the message is borked. We should not add everything
```diff
diff --git a/iterator.go b/iterator.go
index 2a32c28..e45f587 100644
--- a/iterator.go
+++ b/iterator.go
@@ -267,7 +267,9 @@ func (RootLookuper) mergeCNAMEAnswer(resp1, resp2 *dns.Msg) *dns.Msg {
})
// TODO: be smarter when merging OPTs
exdns.ForEachRR(resp2.Extra, func(rr dns.RR) {
-               resp.Ns = append(resp.Extra, rr)
+               if rr.Header().Rrtype != dns.TypeOPT {
+                       resp.Ns = append(resp.Extra, rr)
+               }
})

return resp
```

**Review by karasz — CHANGES_REQUESTED** [review]

File: iterator.go (hunk 1 above)
Comment (karasz) [reply iterator.go:271]: on second thoughts, we only want A/AAAA

**Comment by amery** [comment]
@karasz amended as suggested

**Review by karasz — APPROVED** [review]

**Comment by amery** [comment]
`darvaza.org/resolver@v0.7.20` tagged

**Pull request merged**